src/data_conn.c

```diff
@@ -1706,10 +1719,38 @@ uzfs_zvol_rebuild_scanner(void *arg)
 						    zinfo->name);
 						goto exit;
 					}
+					mutex_enter(&zv->rebuild_mtx);
+					if (zinfo->is_snap_inprogress) {
+						sleep(1);
+						LOG_INFO("waiting for snapshot"
+						    "to finish");
+						mutex_exit(&zv->rebuild_mtx);
+						goto retry;
```

Comment:
can this be just for loop waiting for is_snap_inprogress rather than processing all other code?

Reply:
ok.. this is to send the newly created snap as well if any.. ok.. looks good

Reply:
comments in the code would be better (at other places also)